unlock-app/src/components/interface/locks/Manage/elements/MetadataCard.tsx

```diff
@@ -87,6 +98,132 @@ const MembershipRenewal = ({
   )
 }
 
+const ChangeManagerModal = ({
+  lockAddress,
+  network,
+  manager,
+  tokenId,
+  onChange,
+}: {
+  lockAddress: string
+  network: number
+  manager: string
+  tokenId: string
+  onChange?: any
+}) => {
+  const { getWalletService } = useAuth()
+  const [isOpen, setIsOpen] = useState(false)
+
+  const {
+    setValue,
+    handleSubmit,
+    watch,
+    formState: { isValid },
+    control,
+  } = useForm<{
+    newManager: string
+  }>({
+    defaultValues: {
+      newManager: '',
+    },
+  })
+
+  const newManager = watch('newManager', manager)
+
+  const setKeyManagerForKey = async (newManager: string) => {
+    const walletService = await getWalletService(network)
+    return walletService.setKeyManagerOf({
+      lockAddress,
+      managerAddress: newManager,
+      tokenId,
+    })
+  }
+
+  const changeManagerMutation = useMutation(setKeyManagerForKey, {
+    onSuccess: () => {
+      if (typeof onChange === 'function') {
+        onChange()
+      }
+      ToastHelper.success('Key Manager updated')
+      setIsOpen(false)
+    },
+  })
+
+  const onSubmit = async ({ newManager }: any) => {
+    await changeManagerMutation.mutateAsync(newManager)
+  }
+
+  const recipientIsManager =
+    newManager?.toLowerCase() === manager?.toLowerCase()
+
+  const fieldDisabled =
+    recipientIsManager || changeManagerMutation.isLoading || !isValid
+
+  useEffect(() => {
+    if (isOpen) {
+      setValue('newManager', '') // reset when modal opens
+    }
+  }, [isOpen, setValue])
+
+  return (
+    <>
+      <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
+        <div className="flex flex-col w-full gap-5">
+          <div className="text-left">
+            <h3 className="text-xl font-semibold text-left text-black-500">
+              Change Key Manager
+            </h3>
+            <span className="text-sm leading-tight text-gray-500">
+              Update the address of the Key Manager.
+            </span>
+          </div>
+          <form className="grid w-full gap-3" onSubmit={handleSubmit(onSubmit)}>
+            <Controller
+              name="newManager"
+              control={control}
+              rules={{
+                required: true,
+                validate: isAddressOrEns,
+              }}
+              render={() => {
+                return (
+                  <>
+                    <AddressInput
+                      label="New Manager"
+                      value={newManager}
+                      disabled={changeManagerMutation.isLoading}
+                      onChange={(value: any) => {
+                        setValue('newManager', value, {
+                          shouldValidate: true,
+                        })
+                      }}
+                    />
+                    {recipientIsManager && (
+                      <span className="text-sm text-red-500">
+                        This recipient is the current Key Manager.
```

Comment:
What is `recipient` ?

Reply:
`Recipient` refers to the address: the idea is to prevent the user to submit the form if the current recipient is the actual `keyManager`

<img width="740" alt="error-recipient" src="https://user-images.githubusercontent.com/20865711/226396807-2e75fce4-0b27-4cd8-a5cd-217663173057.png">